test/e2e/v1alpha2/run-e2e-experiment.go

```diff
@@ -0,0 +1,84 @@
+package main
+
+import (
+	"bytes"
+	"io/ioutil"
+	"log"
+	"os"
+	"time"
+
+	k8syaml "k8s.io/apimachinery/pkg/util/yaml"
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	experimentsv1alpha2 "github.com/kubeflow/katib/pkg/api/operators/apis/experiment/v1alpha2"
+	"github.com/kubeflow/katib/pkg/util/v1alpha2/katibclient"
+)
+
+const (
+	timeout = 20 * time.Minute
+)
+
+func main() {
+	if len(os.Args) != 2 {
+		log.Fatal("Experiment name is missing")
+	}
+	expName := os.Args[1]
+	b, err := ioutil.ReadFile(expName)
+	if err != nil {
+		log.Fatal("Error in reading file ", err)
+	}
+	exp := &experimentsv1alpha2.Experiment{}
+	buf := bytes.NewBufferString(string(b))
+	if err = k8syaml.NewYAMLOrJSONDecoder(buf, 1024).Decode(exp); err != nil {
+		log.Fatal("Yaml decode error ", err)
+	}
+	kclient, err := katibclient.NewClient(client.Options{})
+	if err != nil {
+		log.Fatal("NewClient for Katib failed: ", err)
+	}
+	var maxtrials int32 = 3
+	var paralleltrials int32 = 2
+	exp.Spec.MaxTrialCount = &maxtrials
+	exp.Spec.ParallelTrialCount = &paralleltrials
+	err = kclient.CreateExperiment(exp)
+	if err != nil {
+		log.Fatal("CreateExperiment from YAML failed: ", err)
+	}
+
+	for endTime := time.Now().Add(timeout); time.Now().Before(endTime); {
+		exp, err = kclient.GetExperiment(exp.Name, exp.Namespace)
+		if err != nil {
+			log.Fatal("Get Experiment error ", err)
+		}
+		if exp.IsCompleted() {
+			log.Printf("Job %v finished", exp.Name)
+			break
+		}
+		log.Printf("Waiting for job %v to finish. [ %v trials running %v succeeded ]", exp.Name, exp.Status.TrialsRunning, exp.Status.TrialsSucceeded)
+		time.Sleep(20 * time.Second)
+	}
+
+	if !exp.IsCompleted() {
+		log.Fatal("Experiment run timed out")
+	}
+
+	if exp.Status.Trials != *exp.Spec.MaxTrialCount {
+		log.Fatal("All trials are not run in the experiment ", exp.Status.Trials, exp.Spec.MaxTrialCount)
+	}
+
+	if exp.Status.TrialsSucceeded != *exp.Spec.MaxTrialCount {
+		log.Fatal("All trials are not successful ", exp.Status.TrialsSucceeded, *exp.Spec.MaxTrialCount)
+	}
+	if len(exp.Status.CurrentOptimalTrial.ParameterAssignments) == 0 {
```

Comment:
Should we check to see if exp.Status.CurrentOptimalTrial is null first?

Reply:
done